src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,325 @@
+                throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(spiBufferSize));
+            }
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            _gpioDevice = gpioController;
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SendCommand(new SetSegmentReMapColorDepth());
+        }
+
+        /// <summary>
+        /// Initializes new instance of Ssd1351 device that will communicate using SPI bus.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system. 
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize) : this(spiDevice, new GpioController(), dataCommandPin, resetPin, spiBufferSize)
+        {
+            //as a default GPIO controller has been used then dispose of it later..
+            _disposeGpioController = true;
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)((_colorSequence == ColorSequence.ABC ? color.B : color.R) >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(color.G >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)((_colorSequence == ColorSequence.ABC ? color.R : color.B) >> 3);
+            return ((byte)(retval >> 8), (byte)(retval & 0xFF));
+        }
+
+        /// <summary>
+        /// Send filled rectangle to the ssd1351 display.
+        /// </summary>
+        /// <param name="color">The color to fill the rectangle with.</param>
+        /// <param name="x">The x co-ordinate of the point to start the rectangle at.</param>
```

Comment:
Is there a range for both the x and y coordinate possible values so that they still fit on the screen? Also, what is the unit of width and height, pixels? Could we add a <remarks> section on the xml docs here to answer this two questions?

Reply:
Have added the information to the updated PR